src/main/java/com/zendesk/maxwell/producer/MaxwellKafkaProducer.java

```diff
@@ -21,18 +21,14 @@
 
 class KafkaCallback implements Callback {
 	public static final Logger LOGGER = LoggerFactory.getLogger(MaxwellKafkaProducer.class);
-	private InflightMessageList inflightMessages;
-	private final MaxwellContext context;
+	private final AbstractAsyncProducer.CallbackCompleter cc;
 	private final BinlogPosition position;
-	private final boolean isTXCommit;
 	private final String json;
 	private final String key;
 
-	public KafkaCallback(InflightMessageList inflightMessages, BinlogPosition position, boolean isTXCommit, MaxwellContext c, String key, String json) {
-		this.inflightMessages = inflightMessages;
-		this.context = c;
+	public KafkaCallback(AbstractAsyncProducer.CallbackCompleter cc, BinlogPosition position, String key, String json) {
```

Comment:
if `callbackCompleter` became `AsyncProduceRequest` and exposed `key`, `json`, and `position`, that'd reduce some duplication here.   But that can be for another day.